src/ResourceManager/Cdn/Commands.Cdn/CustomDomain/NewAzureRmCdnCustomDomain.cs

```diff
@@ -0,0 +1,94 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using System;
+using System.Management.Automation;
+using System.Net;
+using Microsoft.Azure.Commands.Cdn.Common;
+using Microsoft.Azure.Commands.Cdn.Helpers;
+using Microsoft.Azure.Commands.Cdn.Models.CustomDomain;
+using Microsoft.Azure.Commands.Cdn.Properties;
+using Microsoft.Azure.Management.Cdn;
+using Microsoft.Azure.Management.Cdn.Models;
+
+namespace Microsoft.Azure.Commands.Cdn.CustomDomain
+{
+    [Cmdlet(VerbsCommon.New, "AzureRmCdnCustomDomain", ConfirmImpact = ConfirmImpact.Low), OutputType(typeof(PSCustomDomain))]
+    public class NewAzureRmCdnCustomDomain : AzureCdnCmdletBase
+    {
+        [Parameter(Mandatory = true, HelpMessage = "Host name of the Azure Cdn CustomDomain name.")]
+        [ValidateNotNullOrEmpty]
+        public string HostName { get; set; }
+
+        [Parameter(Mandatory = true, HelpMessage = "Azure Cdn CustomDomain name.")]
+        [ValidateNotNullOrEmpty]
+        public string CustomDomainName { get; set; }
+
+        [Parameter(Mandatory = true, HelpMessage = "Azure Cdn endpoint name.")]
+        [ValidateNotNullOrEmpty]
+        public string EndpointName { get; set; }
+
+        [Parameter(Mandatory = true, HelpMessage = "Azure Cdn profile name.")]
+        [ValidateNotNullOrEmpty]
+        public string ProfileName { get; set; }
+
+        [Parameter(Mandatory = true, HelpMessage = "The resource group of the Azure Cdn Profile")]
+        [ValidateNotNullOrEmpty]
+        public string ResourceGroupName { get; set; }
+
+        public override void ExecuteCmdlet()
+        {
+            try
+            {
+                CdnManagementClient.CustomDomains.GetWithHttpMessagesAsync(
```

Comment:
Our REST doesn't support this yet.
I think "if-none-match" is a improvement we should do...
I believe we have same problem with traffic manager https://github.com/Azure/azure-powershell/blob/dev/src/ResourceManager/TrafficManager/Commands.TrafficManager2/Profile/NewAzureTrafficManagerProfile.cs#L75